src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -629,8 +673,92 @@ subroutine CanopyFluxes(bounds,  num_exposedvegp, filter_exposedvegp,
+         dhsdt_canopy(p) = 0._r8
+         eflx_sh_stem(p) = 0._r8
       end do
 
+      ! calculate biomass heat capacities
+      do f = 1, fn
+         p = filterp(f)
+
+         if(use_biomass_heat_storage) then
+
+            ! fraction of stem receiving incoming radiation
+            frac_rad_abs_by_stem(p) = (esai(p))/(elai(p)+esai(p))
+
+            ! when elai = 0, do not multiply by k_vert (i.e. frac_rad_abs_by_stem = 1)
+            if(elai(p) > 0._r8) frac_rad_abs_by_stem(p) = k_vert * frac_rad_abs_by_stem(p)
+
+            ! if using Satellite Phenology mode, use values in parameter file
+            ! otherwise calculate dbh from stem biomass
+            if(use_cn) then
+               dbh(p) = 2._r8 * sqrt(stem_biomass(p) * (1._r8 - fbw(patch%itype(p))) &
+                    / ( shr_const_pi * htop(p) * k_cyl_vol & 
+                    * nstem(patch%itype(p)) *  wood_density(patch%itype(p))))
+            else
+               dbh(p) = dbh_param(patch%itype(p))
+            endif
+
+            ! leaf and stem surface area
+            sa_leaf(p) = elai(p)
+            ! double in spirit of full surface area for sensible heat
+            sa_leaf(p) = 2.*sa_leaf(p)
+
+            sa_stem(p) = nstem(patch%itype(p))*(htop(p)*shr_const_pi*dbh(p))
+            ! adjust for departure of cylindrical stem model
+            sa_stem(p) = k_cyl_area * sa_stem(p)
+
+            ! do not calculate separate leaf/stem heat capacity for grasses
+            ! or other pfts if dbh is below minimum value
+            if(patch%itype(p) > 11 .or. dbh(p) < min_stem_diameter) then
+               frac_rad_abs_by_stem(p) = 0.0
+               sa_stem(p) = 0.0
+            endif
+            
+            ! cross-sectional area of stems
+            carea_stem = shr_const_pi * (dbh(p)*0.5)**2
+
+            ! if using Satellite Phenology mode, calculate leaf and stem biomass
+            if(.not. use_cn) then
+               ! boreal needleleaf lma*c2b ~ 0.25 kg dry mass/m2(leaf)
+               leaf_biomass(p) = 0.25_r8 * max(0.01_r8, sa_leaf(p)) &
+                    / (1.-fbw(patch%itype(p)))
+               stem_biomass(p) = carea_stem * htop(p) * k_cyl_vol &
+                    * nstem(patch%itype(p)) * wood_density(patch%itype(p)) &
+                    /(1.-fbw(patch%itype(p)))
+            endif
+          
+            ! internal longwave fluxes between leaf and stem
+            ! (use same area of interaction i.e. ignore leaf <-> leaf)
+            sa_internal(p) = min(sa_leaf(p),sa_stem(p))
+            sa_internal(p) = k_internal * sa_internal(p)
+
+            ! calculate specify heat capacity of vegetation
+            ! as weighted averaged of dry biomass and water
+            ! lma_dry has units of kg dry mass/m2 here (table 2 of bonan 2017) 
+            ! cdry_biomass = 1400 J/kg/K, cwater = 4188 J/kg/K
+            cp_leaf(p)  = leaf_biomass(p) * (1400._r8*(1.-fbw(patch%itype(p))) + (fbw(patch%itype(p)))*4188._r8)
```

Comment:
It makes sense that cwater=4188 should go into clm_varcon. If you think cdry_biomass=1400 might need to vary, it probably should go into the params file. But, it would be easier to start with it just being on clm_varcon.

[ ] Make cwater a parameter in clm_varcon
[ ] Make cdry_biomass a parameter either in clm_varcon or on params file.

Reply:
@swensosc can you take this one?

Reply:
Sean put these both in clm_varcon.